metrics/web/docs/features/deep_links/diagrams/parsing_deep_links_class_diagram.puml

```diff
@@ -0,0 +1,48 @@
+@startuml parsing_deep_links_class_diagram
+
+package common.presentation.navigation {
+    class MetricsRouteInformationParser {
+        - _routeConfigurationFactory : RouteConfigurationFactory
+
+        + parseRouteInformation(information: RouteInformation) : RouteConfiguration
+        + restoreRouteInformation(configuration: RouteConfiguration) : RouteInformation
+    }
+
+    package route_configuration {
+        class RouteConfiguration {
+            + name : RouteName
+            + path : String
+            + authorizationRequired : bool
+            + parameters : Map<String, dynamic>
+
+            - RouteConfiguration._(name: RouteName, path: String, authorizationRequired: bool, parameters: Map<String, dynamic> parameters) : RouteConfiguration
+            + RouteConfiguration.dashboard(parameters: Map<String, dynamic>): RouteConfiguration
+            + RouteConfiguration.projectGroups(parameters: Map<String, dynamic>): RouteConfiguration
+            + RouteConfiguration.login(parameters: Map<String, dynamic>): RouteConfiguration
+            + RouteConfiguration.debugMenu(parameters: Map<String, dynamic>): RouteConfiguration
+            + RouteConfiguration.loading(parameters: Map<String, dynamic>): RouteConfiguration
```

Comment:
:readable: Would it be great to mark these methods as `static` to underline that they are constructors?